packages/core/src/agent/MessageSender.ts

```diff
@@ -250,7 +252,55 @@ export class MessageSender {
     }
 
     const outboundPackage = await this.packMessage({ message, keys, endpoint: service.serviceEndpoint })
-    await this.outboundTransporter.sendMessage(outboundPackage)
+    outboundPackage.endpoint = service.serviceEndpoint
+    for (const transport of this.outboundTransporters) {
+      if (transport.supportedSchemes.includes(service.protocolScheme)) {
+        await transport.sendMessage(outboundPackage)
+        break
+      }
+    }
+  }
+
+  private async retrieveServicesByConnection(
+    connection: ConnectionRecord,
+    transportPriority?: TransportPriorityOptions
+  ) {
+    this.logger.debug(
+      `Retrieving services for connection '${connection.id}'${' (' && connection.theirLabel && ')'}${
+        (transportPriority?.restrictive && ' restrictively ',
+        transportPriority && ' by priority of the follow schemes: ' + transportPriority.schemes)
+      }`
+    )
+    // Retrieve DIDComm services
+    const allServices = this.transportService.findDidCommServices(connection)
+
+    //Separate queue service out
+    const services = allServices.filter((s) => !isDidCommTransportQueue(s.serviceEndpoint))
+    const queueService = allServices.find((s) => isDidCommTransportQueue(s.serviceEndpoint))
+
+    //If restrictive will remove services not listed in schemes list
+    if (transportPriority?.restrictive) {
+      services.filter((service) => {
+        const serviceSchema = service.protocolScheme
+        return transportPriority.schemes.includes(serviceSchema)
+      })
+    }
+
+    //If transport priority is set we will sort services by our priority
+    if (transportPriority?.schemes) {
+      services.sort(function (a, b) {
+        const aScheme = a.protocolScheme
+        const bScheme = b.protocolScheme
+        return transportPriority?.schemes.indexOf(aScheme) - transportPriority?.schemes.indexOf(bScheme)
+      })
+    }
+
+    this.logger.debug(
+      `Retrieved ${services.length} services for message to connection '${connection.id}'${
+        ' (' && connection.theirLabel && ')'
```

Comment:
I think this will still just render the value of theirLabel so `(undefined)`. So we can just do this I think:
```
'(${connection.theirLabel})'
```